src/common/thrift/ThriftClientManager-inl.h

```diff
@@ -71,9 +73,16 @@ std::shared_ptr<ClientType> ThriftClientManager<ClientType>::client(const HostAd
 
   VLOG(2) << "Connecting to " << host << " for " << ++connectionCount << " times";
   std::shared_ptr<folly::AsyncSocket> socket;
-  evb->runImmediatelyOrRunInEventBaseThreadAndWait([&socket, evb, resolved]() {
-    socket =
-        folly::AsyncSocket::newSocket(evb, resolved.host, resolved.port, FLAGS_conn_timeout_ms);
+  if (enableSSL_) {
+    socket = folly::AsyncSSLSocket::newSocket(nebula::createSSLContext(), evb);
+  }
+  evb->runImmediatelyOrRunInEventBaseThreadAndWait([this, &socket, evb, resolved]() {
+    if (enableSSL_) {
+      socket->connect(nullptr, resolved.host, resolved.port, FLAGS_conn_timeout_ms);
+    } else {
+      socket =
+          folly::AsyncSocket::newSocket(evb, resolved.host, resolved.port, FLAGS_conn_timeout_ms);
+    }
```

Comment:
why do we need to create the socket inside `runImmediatelyOrRunInEventBaseThreadAndWait`? why not do this like following logics:

```c++
    if (enableSSL_) {
      socket = folly::AsyncSSLSocket::newSocket(nebula::createSSLContext(), evb);
      socket->connect(nullptr, resolved.host, resolved.port, FLAGS_conn_timeout_ms);
    } else {
      socket =  folly::AsyncSocket::newSocket(evb, resolved.host, resolved.port, FLAGS_conn_timeout_ms);
    }
```

Reply:
Good catch. Fixed.